src/main/java/org/elasticsearch/gateway/local/LocalGatewayAllocator.java

```diff
@@ -421,6 +413,25 @@ public boolean apply(DiscoveryNode node) {
         return shardStates;
     }
 
+    private void logListActionFailures(MutableShardRouting shard, String actionType, FailedNodeException[] failures) {
+        if (failures == null || failures.length == 0) {
+            return;
+        }
+        int totalWarnFailures = 0;
+        for (final FailedNodeException failure : failures) {
+            Throwable cause = ExceptionsHelper.unwrapCause(failure);
+            if (cause instanceof ConnectTransportException) {
+                continue;
+            }
+            // we log warn here. debug logs with full stack traces will be logged if debug logging is turned on for TransportNodeListGatewayStartedShards
+            logger.warn("{}: failed to list shard {} on node [{}], reason [{}]", shard.shardId(), actionType, failure.nodeId(), failure.getMessage());
+            totalWarnFailures++;
+        }
+        if (totalWarnFailures > 0) {
+            logger.warn("{}: [{}] failures when trying to list shard {} on nodes.", shard.shardId(), totalWarnFailures, actionType);
+        }
```

Comment:
I think this total message is not needed, since the previous ones are in WARN, so we have an indication already, we just repeat it...